Reorganize packages and models in Devel branch

JunyungKim opened the pull request:
What issue does this change request address?
#49 

What are the significant changes in functionality due to this change request?
0. This pull request covers HTGR models/examples and a part of BoP models/examples
(NHES.Systems.Examples.Nominal_Reactor.HTGR_BOP)
(NHES.Systems.BalanceOfPlant.RankineCycle)
(NHES.Systems.PrimaryHeatSystem.HTGR)

1. Some regression tests are failed in Dymola 2023x. They are working in Dymola 2021x and Dymola 2022 though.
 => FMI_Fluid_CS
 => BOP_OpenFeedHeat_DivertPowerControl_Test

2. NHES.Systems.PrimaryHeatSystem.HTGR.HTGR_Rankine.Examples.Rankine_HTGR_Test  --> deleted.
 => This model does not work in Dymola2021x, Dymola2022, and Dymola2023
 => Almost same model exists. 
       NHES.Systems.PrimaryHeatSystem.HTGR.HTGR_Rankine.Examples.Rankine_HTGR_Test_Transient

2, Some package names and model names are changed in order to have consistency with other packages and models.

3. Examples with models of reactor and BOP are moved to NHES.Systems.Examples
=> For example, 
        NHES.Systems.PrimaryHeatSystem.HTGR.HTGR_Rankine.Examples.Rankine_HTGR_ThreeStageTurbine_Transient
        ==> NHES.Systems.Examples.Nominal_Reactor.HTGR_BOP.HTGR_BOP_L3_Transient

4. Data packages of boundary conditions and initial conditions are integrated into one data package. 

5. Control system models are changed.

Commit:
HTGR package is now on cleaning. Data package integration in HTGR package should be done.

Commit:
HTGR_BraytonCycle constrol system data package is now in the middle of integration.

Commit:
HTGR brayton and rankine cylce package cleaning is almost done. Data package integration is completed.

Commit:
HTGR package cleaning is completed. Further cleaning after discussing with workpackage members can be placed.

Commit:
HTGR package order is changed following Sarah's suggestion.

Commit:
control system cleansing is in progress.

Commit:
control systems of HTGR are in the middle of cleaning.

Commit:
HTGR control systems are almost cleaned.

Commit:
HTGR package clearning is almost done.

Commit:
HTGR package is cleaned. Documentations of example files are added as well.

Commit:
Example model of HTGR is slightly updated so that users can easily figure out which variables need to be tuned. One control system of BOP-L3 is updated.

Commit:
NHES.Systems.EnergyStorage and NHES.Systems.IndustrialProcess are in the middle of cleaning.

mikkdm commented on Models/NHES/Systems/BalanceOfPlant/RankineCycle.mo:
Package needs to be changed to not save files in the package as one file, and those changes reuploaded.

Commit:
obsolete models are deleted in Examples. Nominal Reactor package is newly created.

Commit:
The following packages has been changed so they save not as one file. Systems/BalanceOfPlant/RankineCycle
Systems/EnergyStorage/Concrete_Solid_Media/SupportComponent
Systems/EnergyStorage/SHS_Two_Tank
Systems/EnergyStorage/SensibleheatStorage/Models
Systems/EnergyStoage/SensibleHeatStorage/SupportComponent
Systems/Examples/HTGR_BOP
Systems/Examples/IntegratedEnergySystem
Systems/IndustrialProcess/HeaderTurbineCombo/Models
Systems/IndustrialProcess/HighTempSteamElectrolysis/ControlSystems
Systems/IndustrialProcess/HighTempSteamElectrolysis/Models
Systems/IndustrialProcess/ReverseosmosisDesalination/Models
Systems/PrimaryHeatSystem/HTGR/BraytonCycle
Systems/PrimaryHeatSystem/HTGR/RankineCycle

mikkdm commented on Models/NHES/Systems/BalanceOfPlant/RankineCycle/BaseClasses.mo:
Needs to be not single file package saving.

Commit:
The issue of package file saved as one file is resolved.

Commit:
integration in progress

Commit:
on going

Commit:
OFWH example works now.

Commit:
CFWH model is now working

Commit:
Test files' path is now in updating.

Commit:
regression tests path updating is in progress.

Commit:
1. Regression tests are regoldened (2021x, 2022, and 2023x). 2. HTGR-OFWH and HTGR-CFWH models are added in the regression tests.

mikkdm reviewed (approved):
Changes are approved. Existing branches and forks will need to be re-based after these significant changes.
